readme.txt
This was created as my Project 2 for Web Server Side Programming during the Spring 2013 quarter at RIT.

Almost all of the code was created by myself other than specific sections of PHP provided by professor Dean Ganskop which are primarily located in LIB_project1.php. Most of professor Ganskop's code involved examples of PHP string sanitization and validation functions.

Most of the images used in this project are placeholders because the project's grading was solely focused on programming.

Requirements for this project included:
	- A customized Content Management System
	- A PHP templating system
	- An admin panel to add new and update existing content
	- An automatically generated RSS feed based on newly added content
	- Functionality to load and parse classmates' and other RSS feeds
	- A contact page that visitors can use to automatically email me
	- The creation and use of PHP classes

old/news.txt
22:17:19 03-28-2013|test|This is a test article.
00:38:51 03-29-2013|test 2|This is another test article.
00:39:05 03-29-2013|test 3|This is yet another news article.
00:39:42 03-29-2013|Lorem Ipsum|Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident,sunt in culpa qui officia deserunt mollit anim id est laborum.
00:40:20 03-29-2013|Section 1.10.32 of "de Finibus Bonorum et Malorum"|Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos quiratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?
00:42:06 03-29-2013|Some more placeholder text!|Sollicitudin sapien Eget integer ultrices pede fames class cursus. Amet neque nisl. Montes, pharetra massa condimentum morbi luctus netus viverra est nascetur, conubia turpis consequat luctus vel nec posuere nascetur massa penatibus, magna ut. Turpis, faucibus duis congue nonummy tempor habitasse, ut maecenas vestibulum est fusce ante inceptos porttitor aliquam, hendrerit ut tempus commodo. Non placerat cursus vivamus. Vestibulum et pretium inceptos, mi per nisi, cras. Habitasse sapien luctus aptent eu et massa nulla. Dictumst posuere. Rhoncus felis sagittis Arcu varius. Ante aliquet euismod. Cum feugiat elementum mus. Dictumst rutrum quam. Posuere semper senectus luctus dictum sociosqu faucibus vivamus non magna tristique mauris amet congue montes nostra vivamus nonummy libero ac fusce platea iaculis, lorem rhoncus vulputate nullam sodales porttitor volutpat. Ipsum tellus leo ligula Mus phasellus proin sagittis nascetur sed curae; ligula sagittis litora aptent senectus Semper augue. Dignissim ante cursus leo sagittis etiam justo mauris iaculis maecenas orci ultricies suscipit netus arcu mi luctus. Facilisi et scelerisque eros hymenaeos consectetuer inceptos etiam quam fringilla quisque ut curabitur congue lacus dis dapibus scelerisque facilisi. Senectus. Lectus blandit. Mus. Curabitur venenatis senectus fames et, risus feugiat bibendum tincidunt bibendum fames dis sed cubilia netus enim cum imperdiet vitae odio quam.
00:45:48 03-29-2013|Some more placeholder text!|Sollicitudin sapien Eget integer ultrices pede fames class cursus. Amet neque nisl. Montes, pharetra massa condimentum morbi luctus netus viverra est nascetur, conubia turpis consequat luctus vel nec posuere nascetur massa penatibus, magna ut. Turpis, faucibus duis congue nonummy tempor habitasse, ut maecenas vestibulum est fusce ante inceptos porttitor aliquam, hendrerit ut tempus commodo. Non placerat cursus vivamus. Vestibulum et pretium inceptos, mi per nisi, cras. Habitasse sapien luctus aptent eu et massa nulla. Dictumst posuere. Rhoncus felis sagittis Arcu varius. Ante aliquet euismod. Cum feugiat elementum mus. Dictumst rutrum quam. Posuere semper senectus luctus dictum sociosqu faucibus vivamus non magna tristique mauris amet congue montes nostra vivamus nonummy libero ac fusce platea iaculis, lorem rhoncus vulputate nullam sodales porttitor volutpat. Ipsum tellus leo ligula Mus phasellus proin sagittis nascetur sed curae; ligula sagittis litora aptent senectus Semper augue. Dignissim ante cursus leo sagittis etiam justo mauris iaculis maecenas orci ultricies suscipit netus arcu mi luctus. Facilisi et scelerisque eros hymenaeos consectetuer inceptos etiam quam fringilla quisque ut curabitur congue lacus dis dapibus scelerisque facilisi. Senectus. Lectus blandit. Mus. Curabitur venenatis senectus fames et, risus feugiat bibendum tincidunt bibendum fames dis sed cubilia netus enim cum imperdiet vitae odio quam.

articles/article1.php
<?php
		  include( "../LIB_project1.php" );
		  include( "../P2_Utils.class.php" );
		  $article_html = get_template();
		  $article_dom = new DOMDocument();
		  @$article_dom->loadHTML($article_html);
		  $article_banner_div = $article_dom->getElementById("bannerDiv");
		  P2_Utils::get_banner2($article_dom,$article_banner_div,1);
		  $content_div = $article_dom->getElementById("contentDiv");
		  // Adds the article title to the page in an h1
		  $article_h1 = $article_dom->createElement("h1");
		  $article_h1_text = $article_dom->createTextNode("test 2");
		  $article_h1->appendChild($article_h1_text);
		  $content_div->appendChild($article_h1);
		  // Adds the article pubDate to the page in an h3
		  $article_date_h3 = $article_dom->createElement("h3");
		  $article_date_h3_text = $article_dom->createTextNode("00:38:51 03-29-2013");
		  $article_date_h3->appendChild($article_date_h3_text);
		  $content_div->appendChild($article_date_h3);
		  // Adds the article pubDate to the page in a p
		  $article_description_p = $article_dom->createElement("p");
		  $article_description_p_text = $article_dom->createTextNode("This is another test article.");
		  $article_description_p->appendChild($article_description_p_text);
		  $content_div->appendChild($article_description_p);
		  // Adds a link to navigate back to news.php
		  $article_back_a = $article_dom->createElement("a");
		  $article_back_a->setAttribute("href","http://people.rit.edu/~saw7456/539/project2/news.php");
		  $article_back_a->setAttribute("alt","Back to the News page");
		  $article_back_a_text = $article_dom->createTextNode("Back to News");
		  $article_back_a->appendChild($article_back_a_text);
		  $content_div->appendChild($article_back_a);
		  echo $article_dom->saveHTML();
		  ?>
